app/database/db.go

```diff
@@ -28,14 +31,45 @@ func Open(source interface{}) (*DB, error) {
 	var err error
 	var dbConn *gorm.DB
 	var driverName = "mysql"
-	dbConn, err = gorm.Open(driverName, source)
+
+	var rawConnection gorm.SQLCommon
+	switch src := source.(type) {
+	case string:
+		rawConnection, err = OpenRawDBConnection(src)
+		if err != nil {
+			return nil, err
+		}
+	case gorm.SQLCommon:
+		rawConnection = src
+	default:
+		return nil, fmt.Errorf("unknown database source type: %T (%v)", src, src)
+	}
+	dbConn, err = gorm.Open(driverName, rawConnection)
 	dbLogger, logMode := logging.NewDBLogger()
 	dbConn.LogMode(logMode)
 	dbConn.SetLogger(dbLogger)
 
 	return newDB(dbConn), err
 }
 
+// OpenRawDBConnection creates a new DB connection
+func OpenRawDBConnection(sourceDSN string) (*sql.DB, error) {
+	logger, _ := logging.NewRawDBLogger()
+	registerDriver := true
+	for _, driverName := range sql.Drivers() {
+		if driverName == "instrumented-mysql" {
+			registerDriver = false
+			break
+		}
+	}
+
+	if registerDriver {
+		sql.Register("instrumented-mysql",
+			instrumentedsql.WrapDriver(&mysql.MySQLDriver{}, instrumentedsql.WithLogger(logger)))
```

Comment:
You set the logger while it is already set in `Open(...)` at `.LogMode` and `.SetLogger`. I think I understand what you try to do but still I don't find this very clean :-/

Reply:
Those are two different loggers, a high-level logger (gorm) and a low level logger (sql)

Reply:
Those are two different loggers, a high-level logger (gorm) and a low level logger (sql)